client/query.go

```diff
@@ -62,35 +62,27 @@ func (c *QueryClient) Query(nrqlQuery string, response interface{}) (err error)
 		return errors.New("go-insights: Invalid query response can not be nil")
 	}
 
-	err = c.generateQueryURL(nrqlQuery)
-	if err != nil {
-		return err
-	}
-
-	c.Logger.Debugf("Querying: %s", nrqlQuery)
-	err = c.queryRequest(response)
-	if err != nil {
-		return err
-	}
+	err = c.queryRequest(nrqlQuery, response)
 
 	return nil
 }
 
 // queryRequest makes a NRQL query and returns the result in `queryResult`
 // which must be a pointer to a struct that the JSON package can unmarshall
-func (c *QueryClient) queryRequest(queryResult interface{}) (err error) {
+func (c *QueryClient) queryRequest(nrqlQuery string, queryResult interface{}) (err error) {
 	var request *http.Request
 	var response *http.Response
 
-	if len(c.URL.RawQuery) < 1 {
-		return fmt.Errorf("Query string can not be empty")
+	queryURL, err := c.generateQueryURL(nrqlQuery)
+	if err != nil {
+		return errors.New("Unable to generate query URL")
```

Comment:
Should return the error from `c.generateQueryURL`, if that error isn't informative, we should fix it in `generateQueryURL`

Reply:
Returning this error now.